libs/src/sdk/api/tracks/TracksApi.ts

```diff
@@ -75,44 +82,102 @@ export class TracksApi extends TracksApiWithoutStream {
       throw new Error('Metadata object is not valid')
     }
 
-    requestParameters.metadata.owner_id = Number(requestParameters.artistId)
+    const artistId = decodeHashId(requestParameters.artistId)
+    if (artistId === null) {
+      throw new Error('artistId is not valid')
+    }
+
+    requestParameters.metadata.owner_id = artistId
 
     const metadataMissingValues = objectMissingValues(
       requestParameters.metadata,
       TRACK_REQUIRED_VALUES
     )
-    if (metadataMissingValues) {
+    if (metadataMissingValues?.length) {
       throw new Error(
         `Metadata object is missing values: ${metadataMissingValues}`
       )
     }
 
     // Upload track audio and cover art to storage node
-    const updatedMetadata = await this.storage.uploadTrackAudioAndCoverArt(
-      trackFile,
-      coverArtFile,
-      metadata,
-      onProgress
-    )
+    const [audioResp, coverArtResp] = await Promise.all([
+      retry3(
+        async () =>
+          await this.storage.uploadFile({
+            file: trackFile,
+            onProgress,
+            template: 'audio'
+          }),
+        (e) => {
+          console.log('Retrying uploadTrackAudio', e)
+        }
+      ),
+      retry3(
+        async () =>
+          await this.storage.uploadFile({
+            file: coverArtFile,
+            onProgress,
+            template: 'img_square'
+          }),
+        (e) => {
+          console.log('Retrying uploadTrackCoverArt', e)
+        }
+      )
+    ])
+
+    // Update metadata to include uploaded CIDs
+    const updatedMetadata = {
+      ...metadata,
+      track_segments: [],
+      track_cid: audioResp.results['320'],
+      download: metadata.download?.is_downloadable
+        ? {
+            ...metadata.download,
+            cid: metadata.track_cid
+          }
+        : metadata.download,
+      cover_art_sizes: coverArtResp.id,
+      duration: parseInt(audioResp.probe.format.duration, 10)
+    }
 
-    // TODO: Integrate with wallet api
-    // TODO: Write metadata to chain
-    // const trackId = await this._generateTrackId()
-    // const response = await this.contracts.EntityManagerClient!.manageEntity(
-    //   ownerId,
-    //   EntityManagerClient.EntityType.TRACK,
-    //   trackId,
-    //   EntityManagerClient.Action.CREATE,
-    //   JSON.stringify({ cid: updatedMetadata.track_cid, data: updatedMetadata })
-    // )
-    // const txReceipt = response.txReceipt
+    // Write metadata to chain
+    const metadataCid = await generateMetadataCidV1(updatedMetadata)
+    const trackId = await this.generateTrackId()
+    const response = await this.entityManager.manageEntity({
+      userId: artistId,
+      entityType: EntityType.TRACK,
+      entityId: trackId,
+      action: Action.CREATE,
+      metadata: JSON.stringify({
+        cid: updatedMetadata.track_cid,
```

Comment:
this was a bug I made earlier - should be changed to `metadataCid.toString()` instead of `track_cid` like [this](https://github.com/AudiusProject/audius-protocol/pull/5175/files#diff-866586d98c7a27e7761e288ddb9c9863059b8d5737899cfd21ffd494ef63148aR430)

Reply:
Ah, oops. Good catch!